src/makeInfinityAdder.js

```diff
@@ -4,7 +4,21 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  return function makeAdder(arg) {
+    if (arg || arg === 0) {
```

Comment:
The check `arg || arg === 0` is incorrect because it will evaluate to `true` for `arg=0`. Since the intention is to check for the presence of an argument, consider using `arg !== undefined` to handle all numbers correctly, including `0`.